Realm/RLMProperty.mm

```diff
@@ -244,16 +244,22 @@ - (instancetype)initSwiftPropertyWithName:(NSString *)name
         _objcRawType = [NSString stringWithFormat:@"@\"RLMArray<%@>\"", [[obj valueForKey:_name] objectClassName]];
     }
 
-    if (![self setTypeFromRawType]) {
+    void (^throwForPropertyName)(NSString *) = ^(NSString *propertyName){
         NSString *reason = [NSString stringWithFormat:@"Can't persist property '%@' with incompatible type. "
-                            "Add to ignoredPropertyNames: method to ignore.", self.name];
+                            "Add to Object.ignoredProperties() class method to ignore.", propertyName];
         @throw RLMException(reason);
+    };
+
+    if (![self setTypeFromRawType]) {
+        throwForPropertyName(self.name);
     }
 
     // convert type for any swift property types (which are parsed as Any)
     if (_type == RLMPropertyTypeAny) {
         if ([[obj valueForKey:_name] isKindOfClass:[NSString class]]) {
             _type = RLMPropertyTypeString;
+        } else if (![[obj valueForKey:_name] isKindOfClass:[RLMListBase class]]) {
```

Comment:
This check is kinda weird because it only works due to a Swift 1.2 bug and it'd break horribly if it was reachable with Swift 2.0. I think it'd be better to check `propArray` for `Any` properties in `+propertiesForClass:isSwift:` after the handling for list properties is done and there shouldn't be any Any properties left.